[JUnit Platform] Enable parallel execution of features (#2604)
cucumber/cucumber-jvm: cucumber-junit-platform-engine/README.md

with this configuration:

```properties
cucumber.execution.exclusive-resources.isolated.read-write=org.junit.platform.engine.support.hierarchical.ExclusiveResource.GLOBAL_KEY
```

> ### Sequential Scenario Execution With Parallel Run

Reviewer: This section and its fails to mention the most important aspect first: JUnit 4 compatibility. Please rewrite with this in mind.

> By default, when the configuration parameter `cucumber.execution.parallel.enabled` set to `true`, all applicable

Reviewer: Consider simplifying this phrase and its duplicates to "when parallel execution is enabled".

> scenarios would be executed in parallel. setting the configuration parameter

Reviewer: ```suggestion
scenarios are executed in parallel. Setting the configuration parameter
```

And there are a few more sentences that don't start with a capital.

`cucumber.execution.execution-mode.scenario` to `same_thread` results in features executed in parallel but scenarios
within the feature runs sequential (parallel run behaviour of junit4). 

Example - consider a test suite of 2 Features

```gherkin
Feature: Sequential Scenario Execution Feature 1
   
   Scenario: scenario 1
      
   Scenario: scenario 2

Feature: Sequential Scenario Execution Feature 2

   Scenario: scenario 1

   Scenario: scenario 2
```
when configuration parameter `cucumber.execution.parallel.enabled` set to `true` and
configuration parameter `cucumber.execution.execution-mode.scenario` set to `same_thread` Then the execution would be

Thread 1 -> Feature 1 -> scenario 1 -> scenario 2
<p>Thread 2 -> Feature 2 -> scenario 1 -> scenario 2

> default value of configuration parameter `cucumber.execution.execution-mode.scenario` is `concurrent` 

Reviewer: This need not be mentioned. It is covered in the `Configuration Options` section of the documentation.

cucumber.execution.dry-run=                                   # true or false.
                                                              # default: false

cucumber.execution.execution-mode.feature=                    # same_thread or concurrent
                                                              # default: concurrent
                                                              # same_thread - executes scenarios sequentially in the 
                                                              # same thread as the parent feature 

>                                                               # conncurrent - executes scenarios concurrently on any

Reviewer 2: ```suggestion
                                                              # concurrent - executes scenarios concurrently on any
```

Reviewer: Cheers. Fixed.

cucumber.execution.parallel.config.custom.class=              # class name.
                                                              # example: com.example.MyCustomParallelStrategy

> cucumber.execution.execution-mode.scenario=                   # same_thread or concurrent

Reviewer: I suggested a `cucumber.execution.execution-mode.feature` because all scenarios in a feature will be executed in the same thread. Is there a reason you've changed this to scenario?

Author: I was thinking that naming the constant to `cucumber.execution.execution-mode.feature` will create a confusion when setting `cucumber.execution.execution-mode.feature=same_thread`, it may lead to users misinterpret that all features will run in the same thread. but below sentence makes sense

> Note that strictly speaking scenarios don't' exist. There are only pickles (both scenarios and scenario outlines are turned into pickles for individual execution).

I will rename the constant to `cucumber.execution.execution-mode.feature`. Thanks for your suggesstion!
